src/relay/pass/fold_scale_axis.cc

```diff
@@ -0,0 +1,559 @@
+
+STuple ReluForwardTransform(const Call& ref_call,
+                              const AxesSet& expected_axes,
+                              const Array<STuple>& sargs) {
+  if (!sargs[0]->axes.defined()) return STuple();
+  // return transformed conv2d
+  auto rnode = make_node<STupleNode>();
+  rnode->value = CallNode::make(
+      ref_call->op, {sargs[0]->value}, ref_call->attrs, {});
+  rnode->scale = sargs[0]->scale;
+  rnode->axes = sargs[0]->axes;
+  return STuple(rnode);
+}
+
+RELAY_REGISTER_OP("nn.relu")
+.set_attr<FForwardPrep>("FScaleAxisForwardPrep", ReluForwardPrep);
+
+RELAY_REGISTER_OP("nn.relu")
+.set_attr<FForwardTransform>("FScaleAxisForwardTransform", ReluForwardTransform);
+
+RELAY_REGISTER_OP("nn.leaky_relu")
+.set_attr<FForwardPrep>("FScaleAxisForwardPrep", ReluForwardPrep);
+
+RELAY_REGISTER_OP("nn.leaky_relu")
+.set_attr<FForwardTransform>("FScaleAxisForwardTransform", ReluForwardTransform);
+
+// AddSub
+Array<AxesSet> AddSubForwardPrep(const Call& call, AxesSet out_axes) {
+  const auto* tlhs = call->args[0]->checked_type().as<TensorTypeNode>();
+  const auto* trhs = call->args[1]->checked_type().as<TensorTypeNode>();
+  auto none = NullValue<AxesSet>();
+  if (MatchBroadcastToLeftAxes(tlhs, trhs, out_axes)) {
+    return {out_axes, none};
+  } else if (MatchBroadcastToLeftAxes(trhs, tlhs, out_axes)) {
+    return {none, out_axes};
+  } else {
+    return {none, none};
+  }
+}
+
+STuple AddSubForwardTransform(const Call& ref_call,
+                              const AxesSet& expected_out_axes,
+                              const Array<STuple>& sargs) {
+  if (!sargs[0]->axes.defined() && !sargs[1]->axes.defined()) {
+    return STuple();
+  }
+  const auto* tlhs = ref_call->args[0]->checked_type().as<TensorTypeNode>();
+  const auto* trhs = ref_call->args[1]->checked_type().as<TensorTypeNode>();
+
+  auto rnode = make_node<STupleNode>();
+
+  if (sargs[0]->axes.defined()) {
+    CHECK(!sargs[1]->axes.defined());
+    CHECK(MatchBroadcastToLeftAxes(tlhs, trhs, sargs[0]->axes));
+    Expr scale = ExpandBiasToMatchAxis(
+        sargs[0]->scale, tlhs->shape.size(), sargs[0]->axes);
+    Expr rhs = Divide(sargs[1]->value, scale);
+    rnode->value = CallNode::make(ref_call->op, {sargs[0]->value, rhs},
+                                  ref_call->attrs, ref_call->type_args);
+    rnode->scale = sargs[0]->scale;
+    rnode->axes = sargs[0]->axes;
+  } else {
+    CHECK(sargs[1]->axes.defined());
+    CHECK(sargs[0]->axes.defined());
+    CHECK(MatchBroadcastToLeftAxes(trhs, tlhs, sargs[1]->axes));
+    Expr scale = ExpandBiasToMatchAxis(
+        sargs[1]->scale, trhs->shape.size(), sargs[1]->axes);
+    Expr lhs = Divide(sargs[0]->value, scale);
+    rnode->value = CallNode::make(ref_call->op, {lhs, sargs[1]->value},
+                                  ref_call->attrs, ref_call->type_args);
+    rnode->scale = sargs[1]->scale;
+    rnode->axes = sargs[1]->axes;
+  }
+  return STuple(rnode);
+}
+
+RELAY_REGISTER_OP("add")
```

Comment:
s/add/addition? To keep consistent to `subtract`, `multiply`, and `divide`

Reply:
This follow numpy convention